src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -1019,6 +1018,46 @@ private function relativize($xpath)
     }
 
     /**
+     * Splits the XPath into parts that are separated by the union operator.
+     *
+     * @param string $xpath
+     *
+     * @return string[]
+     */
+    private function splitUnionParts($xpath)
+    {
+        // Split any unions into individual expressions. We need to iterate
+        // through the string to correctly parse opening/closing quotes and
+        // braces which is not possible with regular expressions.
+        $unionParts = array();
+        $inSingleQuotedString = false;
+        $inDoubleQuotedString = false;
+        $openedBrackets = 0;
+        $lastUnion = 0;
+        for ($i = 0; $i < strlen($xpath); ++$i) {
```

Comment:
`strlen()` should be moved out from, from here to not be called on every interation of `for()`